### PR TITLE
[Data Object] Relations should be always lazy loaded, remove config o…

### LINKED ISSUE #5772: [Data Object] Relations should be always lazy loaded, remove config option in definition

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #5772

### CODE REVIEW

File: models/DataObject/Traits/FieldcollectionObjectbrickDefinitionTrait.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @category   Pimcore
+ * @package    Element
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model\DataObject\Traits;
+
+trait FieldcollectionObjectbrickDefinitionTrait
```
Comment (brusch) [inline models/DataObject/Traits/FieldcollectionObjectbrickDefinitionTrait.php]: That's not related to this PR, right?

File: models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php [hunk 2]
```diff
@@ -74,20 +69,9 @@ public function setClasses($classes)
...
-        return $this->lazyLoading;
+        return true;
     }
 
-    /**
-     * @param bool $lazyLoading
-     *
-     * @return $this
-     */
-    public function setLazyLoading($lazyLoading)
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: still need this? https://github.com/pimcore/pimcore/blob/fb48e8b95b53fa0cd8efa33de36bea01705210c5/models/DataObject/ClassDefinition/Data/ReverseManyToManyObjectRelation.php#L79

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: No, didn't have any effect anyway. 👍

File: models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php [hunk 3]
```diff
@@ -74,20 +69,9 @@ public function setClasses($classes)
      */
     public function getLazyLoading()
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: please mark deprecated if removed in 7

Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: only `method_exists('getLazyLoading')` calls will be removed

**Pull request merged**